tests/py/unit/test_hot_reload.py

```
@@ -0,0 +1,191 @@
+
+class TestHotReload:
+    """Tests for the HotReload class."""
+
+    def test_creates_session_registry(self) -> None:
+        """HotReload has a session registry."""
+        from trellis.utils.hot_reload import HotReload, SessionRegistry
+
+        hr = HotReload()
+        assert isinstance(hr.sessions, SessionRegistry)
+
+    def test_invalidate_all_sessions_marks_all_elements_dirty(self, noop_component) -> None:
```

Ruff (0.14.10)

Unused method argument: `noop_component`

(ARG002)